pkg/controller/controller_binding.go

```diff
@@ -365,13 +365,13 @@ func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[st
 	}
 
 	found := false
-
 	_, err := c.kubeClient.Core().Secrets(binding.Namespace).Get(binding.Spec.SecretName, metav1.GetOptions{})
 	if err == nil {
 		found = true
 	}
 
 	if found {
+		// TODO: Check ownerReferences first?
 		_, err = c.kubeClient.Core().Secrets(binding.Namespace).Update(secret)
 	} else {
 		_, err = c.kubeClient.Core().Secrets(binding.Namespace).Create(secret)
```

Comment:
To better handle concurrent events this whole get+create/update should be in a for loop with retries on "conflict" for update and "already exists" for create. Also other controllers use listers (i.e. Store) from informers for reads to avoid hitting api server directly. This can/should only be done if there is a mechanism already that will trigger reprocessing on Secret event (update/etc). But I'm not familiar enough with the code to suggest to do that :)